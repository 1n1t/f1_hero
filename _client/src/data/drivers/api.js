/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getDriverList: '/api/drivers'
};

export const getDriverList = () => fetchApi(endPoints.getDriverList);
