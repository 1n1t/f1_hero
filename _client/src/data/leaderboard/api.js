/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getUsersScores: '/api/leaderboard'
};

export const getUsersScores = () => fetchApi(endPoints.getUsersScores);
