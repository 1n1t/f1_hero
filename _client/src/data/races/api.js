/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getRaceList: '/api/races'
};

export const getRaceList = () => fetchApi(endPoints.getRaceList);
