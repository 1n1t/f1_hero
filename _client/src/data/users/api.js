/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

const endPoints = {
	getCurrentUser: '/auth/current-user'
};

export const getCurrentUser = () => fetchApi(endPoints.getCurrentUser);
