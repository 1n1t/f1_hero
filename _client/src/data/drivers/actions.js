/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateDriverList = drivers => ({
	type: actionTypes.UPDATE_DRIVER_LIST,
	payload: drivers
});

export const getDriverList = () => async dispatch => {
	const response = await api.getDriverList();
	dispatch(updateDriverList(response.data.drivers));
};
