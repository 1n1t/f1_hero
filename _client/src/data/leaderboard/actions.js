/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

export const updateScores = scores => ({
	type: actionTypes.UPDATE_SCORES,
	payload: scores
});

export const getUsersScores = () => async dispatch => {
	const response = await api.getUsersScores();
	dispatch(updateScores(response.data.leaderboard));
};
