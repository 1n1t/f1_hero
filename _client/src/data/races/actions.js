/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateRaceList = races => ({
	type: actionTypes.UPDATE_RACE_LIST,
	payload: races
});

export const selectRace = id => ({
	type: actionTypes.SELECT_RACE,
	payload: id
});

export const updateRace = ({ round, field, value }) => ({
	type: actionTypes.UPDATE_RACE,
	payload: { round, field, value }
});

export const getRaceList = () => async dispatch => {
	const response = await api.getRaceList();
	dispatch(updateRaceList(response.data.races));
};
