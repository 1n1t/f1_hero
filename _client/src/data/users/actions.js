/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

export const updateCurrentUser = user => ({
	type: actionTypes.UPDATE_CURRENT_USER,
	payload: user
});

export const getCurrentUser = () => async dispatch => {
	const response = await api.getCurrentUser();
	dispatch(updateCurrentUser(response.data.user));
};
