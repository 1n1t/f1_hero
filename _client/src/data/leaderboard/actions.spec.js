import leaderboardSample from 'JestClient/samples/leaderboard';
import t from 'JestClient/actionTester';
import { getUsersScores, updateScores } from './actions';
import { endPoints } from './api';

const { scores } = leaderboardSample;

it('dispatches "leaderboard/UPDATE_SCORES" action with correct drivers', () => {
	expect(updateScores(scores)).toMatchSnapshot();
});

it('fetches data from api and create correct action creator', done => {
	const dispatchMock = jest.fn();

	const assertions = (dispatch, request) => {
		expect(request.url).toEqual(endPoints.getUsersScores);
		expect(dispatch).toHaveBeenCalledTimes(1);
		expect(dispatch).toHaveBeenCalledWith(updateScores(scores));
		done();
	};

	t({
		action: getUsersScores,
		apiResponse: { leaderboard: scores },
		dispatch: dispatchMock,
		assertions
	});
});
