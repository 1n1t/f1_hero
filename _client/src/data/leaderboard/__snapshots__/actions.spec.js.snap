// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatches "leaderboard/UPDATE_SCORES" action with correct payload 1`] = `
Object {
  "payload": Array [
    Object {
      "total": 123,
      "user": "John Doe",
    },
    Object {
      "total": 95,
      "user": "Mary Doe",
    },
    Object {
      "total": 12,
      "user": "Josh Simpson",
    },
  ],
  "type": "leaderboard/UPDATE_SCORES",
}
`;
